oxenstored/connections.ml

```diff
@@ -97,39 +97,50 @@ let del_watches cons (con : Connection.t) =
       )
   )
 
-let del_anonymous cons con spec_fds =
+(* Reallocate the poll_status array, update indices pointing to it *)
+let resize_anonymous cons spec_fds =
+  cons.poll_status <-
+    Array.make
+      (Hashtbl.length cons.anonymous + List.length spec_fds)
+      (default_poll_status ())
+  (* TODO: Work around an unnecessary allocation here *) ;
+
+  (* Keep the special fds at the beginning *)
+  let i =
+    List.fold_left
+      (fun index fd ->
+        cons.poll_status.(index) <- (fd, spec_poll_status ()) ;
+        index + 1
+      )
+      0 spec_fds
+  in
+
+  let _ =
+    Hashtbl.fold
+      (fun key (con, _) i ->
+        Hashtbl.replace cons.anonymous key (con, i) ;
+        (* SAFETY: con.poll_status is always Some for anonymous connections *)
+        cons.poll_status.(i) <-
+          (Connection.get_fd con, Option.get con.poll_status) ;
```

Comment:
There's a way to completely avoid exceptions:
```suggestion
        Option.iter
          (fun status -> cons.poll_status.(i) <- (Connection.get_fd con, status))
          con.poll_status
```

Reply:
I think my comment from last time still applies - this logically shouldn't happen and I would rather see an exception than a non-updated poll_status if it does.

Reply:
What does the exception provoke? xenstore exiting and restarting?

Reply:
I agree with @last-genius; this is basically an assertion. My only concern is: are we able to locate the failure if it happens or do we want to log this before we fail?

Reply:
If this is a one-off problem, there might be no opportunity to re-run with `OCAMLRUNPARAM=b` - hence it's important to capture this failure when it happens.